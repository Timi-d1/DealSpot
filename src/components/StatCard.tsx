import React from 'react';
import styles from './Dashboard.module.css';

interface StatCardProps {
  icon: string;
  label: string;
  value: string;
}

const StatCard: React.FC<StatCardProps> = ({ icon, label, value }) => {
  return (
    <div className={styles.statCard}>
      <div className={styles.statIcon}  data-testid="stat-icon">
        <div dangerouslySetInnerHTML={{ __html: icon }} />
      </div>
      <div className={styles.statLabel}>{label}</div>
      <div className={styles.statValue}>{value}</div>
    </div>
  );
};

export default StatCard;