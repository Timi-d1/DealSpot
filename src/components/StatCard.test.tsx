import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import StatCard from './StatCard';

describe('StatCard Component', () => {
  const mockProps = {
    icon: '<svg><circle cx="10" cy="10" r="10" /></svg>',
    label: 'Total Sales',
    value: '1000',
  };

  it('renders the label correctly', () => {
    render(<StatCard {...mockProps} />);
    expect(screen.getByText('Total Sales')).toBeInTheDocument();
  });

  it('renders the value correctly', () => {
    render(<StatCard {...mockProps} />);
    expect(screen.getByText('1000')).toBeInTheDocument();
  });

  it('renders the icon HTML safely using dangerouslySetInnerHTML', () => {
    render(<StatCard {...mockProps} />);
    const iconContainer = screen.getByTestId('stat-icon');
    expect(iconContainer.innerHTML).toContain('<svg>');
  });

  it('matches snapshot', () => {
    const { container } = render(<StatCard {...mockProps} />);
    expect(container).toMatchSnapshot();
  });
});
