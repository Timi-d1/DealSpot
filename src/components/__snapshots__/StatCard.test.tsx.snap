// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatCard Component matches snapshot 1`] = `
<div>
  <div
    class="statCard"
  >
    <div
      class="statIcon"
      data-testid="stat-icon"
    >
      <div>
        <svg>
          <circle
            cx="10"
            cy="10"
            r="10"
          />
        </svg>
      </div>
    </div>
    <div
      class="statLabel"
    >
      Total Sales
    </div>
    <div
      class="statValue"
    >
      1000
    </div>
  </div>
</div>
`;
